File: packages/api/src/models/Supplier.ts
import mongoose, { Schema, Document, Model } from 'mongoose';
import { Supplier, QualificationFile, ContactPerson, SupplierQualificationStatus } from '@app/shared';

const ContactPersonSchema = new Schema<ContactPerson>({
  name: { type: String, required: true },
  title: String,
  phone: { type: String, required: true },
  email: { type: String, required: true }
}, { _id: false });

const QualificationFileSchema = new Schema<QualificationFile>({
  id: { type: String, required: true },
  name: { type: String, required: true },
  type: { type: String, required: true },
  issueDate: { type: Date, required: true },
  expiryDate: { type: Date, required: true },
  status: { type: String, enum: ['valid', 'expiring', 'expired'], default: 'valid' },
  attachmentId: { type: String, required: true }
}, { _id: false });

const SupplierSchema = new Schema<Supplier & Document>({
  code: { type: String, required: true, unique: true, index: true },
  name: { type: String, required: true },
  shortName: String,
  category: [String],
  businessLicense: String,
  contactPerson: ContactPersonSchema,
  address: String,
  bankAccount: String,
  qualifications: [QualificationFileSchema],
  qualificationStatus: { 
    type: String, 
    enum: ['qualified', 'warning', 'expired', 'blacklisted', 'pending'],
    default: 'pending' as SupplierQualificationStatus 
  },
  rating: { type: Number, default: 0, min: 0, max: 5 },
  tags: [String],
  registeredAt: { type: Date, default: Date.now }
}, {
  timestamps: true
});

SupplierSchema.index({ name: 'text', shortName: 'text', code: 'text' });
SupplierSchema.index({ qualificationStatus: 1 });
SupplierSchema.index({ 'qualifications.expiryDate': 1 });

export const SupplierModel: Model<Supplier & Document> = 
  mongoose.model('Supplier', SupplierSchema);

File: backend/MedicalAllocation.Domain/Entities/Supplier.cs
namespace MedicalAllocation.Domain.Entities;

public class Supplier
{
    public int Id { get; set; }
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string ContactPerson { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Address { get; set; } = string.Empty;
    public string BusinessLicense { get; set; } = string.Empty;
    public string GspCertificate { get; set; } = string.Empty;
    public int Rating { get; set; } = 3;
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime? UpdatedAt { get; set; }
}
